### ISSUE TITLE
Update desktop Places to use application-services places component

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/uniffi-rs#626]
**TL;DR Why:** Integrate the places (bookmarks and history) rust component to desktop to update the current code base with more up to date, re-usable, and easier to maintain code. The rust upgrade can also help us build on top of places at a quicker rate.

From a product standpoint, the new rust code implementation of bookmarks in desktop would allow the bookmarks team to change some of the UI bookmarks experience and modernize it. The new infrastructure and architecture will allow them to modify and experiment with bookmarks. As a team, with the new implementation we can run experiments to identify areas we can increase engagement with synced and local bookmarks.

**Comment by data-sync-user** [comment]
➤ Ana Medinac commented:

Later.  We could focus on upgrading places component after FxA.
